app/helpers/api_page_helper.rb

```diff
@@ -0,0 +1,50 @@
+module ApiPageHelper
+  # ask for current_weather
+  def get_weather(city = 'berlin', api_id)
+    # Rails.cache.clear
+    if api_id == 1
+      cache_name = 'curr_weather'
+    elsif api_id == 2
+      cache_name = 'forecast_weather'
+    end
+
+    exists_weather_of(city, api_id, cache_name)
+    if api_id == 1
+      get_curr_weather_data(Rails.cache.fetch('curr_weather'))
+    elsif api_id == 2
+      get_forecast_weather_data(Rails.cache.fetch('forecast_weather'))
+    end
+  end
+
+  def exists_weather_of(city, api_id, cache_name)
+    if Rails.cache.fetch(cache_name)
+      test = Rails.cache.fetch(cache_name)
+      if test['name'] == city.capitalize
```

Comment:
Ahh so you're seeing if information is city specific based on what's stored already?